lopper/assists/openamp_xlnx.py

```diff
@@ -1074,12 +1076,161 @@ def xlnx_remoteproc_construct_carveouts(tree, carveouts, new_ddr_nodes, verbose
             print("WARNING: invalid remoteproc elfload carveout", carveout)
             return False
 
+    openamp_channel_info["new_ddr_nodes"+channel_id] = new_ddr_nodes
+
+    return True
+
+def platform_validate(platform):
+    if platform not in [ SOC_TYPE.ZYNQMP, SOC_TYPE.VERSAL, SOC_TYPE.VERSAL_NET ]:
+        print("ERROR: unsupported platform: ", platform)
+        return False
+    return True
+
+def xlnx_remoteproc_v2_construct_cluster(tree, channel_id, openamp_channel_info, verbose = 0):
+    platform = get_platform(tree, verbose)
+    cpu_config = openamp_channel_info["cpu_config"+channel_id]
+    node = openamp_channel_info["node"+channel_id]
+    host_node = node.parent.parent
+    remote_node = openamp_channel_info["remote_node"+channel_id]
+    elfload_nodes = openamp_channel_info["elfload"+channel_id]
+    new_ddr_nodes = openamp_channel_info["new_ddr_nodes"+channel_id]
+    cluster_node = None
+    rpu_core = openamp_channel_info["rpu_core"+channel_id]
+    core_index = openamp_channel_info["core_index"]
+    rpu_core = determinte_rpu_core(tree, cpu_config, remote_node )
+    rpu_core_pd_prop = openamp_channel_info["rpu_core_pd_prop"+channel_id]
+    core_compat_substr = "tcm-1.0"
+    absolute_addr_tcm_substr = "tcm-global-1.0"
+    core_reg_val = []
+    core_reg_names = []
+    core_reg_names_base_str = "tcm0"
+    core_reg_names_index = 0
+    power_domains = []
+    cluster_ranges_val = []
+
+    cluster_reg = 0xff9a0000
```

Comment:
cluster reg isn't in upstream binding I think.

Reply:
done.